identity/src/test/java/com/android/identity/MobileSecurityObjectGeneratorTest.java

```diff
@@ -0,0 +1,315 @@
+        Assert.assertEquals("D80B83D25173C484C5640610FF1A31C949C1D934BF4CF7F18D5223B15DD4F21C"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(0L)));
+        Assert.assertEquals("4D80E1E2E4FB246D97895427CE7000BB59BB24C8CD003ECF94BF35BBD2917E34"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(1L)));
+        Assert.assertEquals("8B331F3B685BCA372E85351A25C9484AB7AFCDF0D2233105511F778D98C2F544"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(2L)));
+        Assert.assertEquals("C343AF1BD1690715439161ABA73702C474ABF992B20C9FB55C36A336EBE01A87"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(3L)));
+    }
+
+    @Test
+    @SmallTest
+    public void testBasicMSO() throws Exception {
+        PublicKey deviceKeyFromVector = Util.getPublicKeyFromIntegers(
+                new BigInteger(TestVectors.ISO_18013_5_ANNEX_D_STATIC_DEVICE_KEY_X, 16),
+                new BigInteger(TestVectors.ISO_18013_5_ANNEX_D_STATIC_DEVICE_KEY_Y, 16));
+        final Timestamp signedTimestamp = Timestamp.ofEpochMilli(1601559002000L);
+        final Timestamp validFromTimestamp = Timestamp.ofEpochMilli(1601559002000L);
+        final Timestamp validUntilTimestamp = Timestamp.ofEpochMilli(1633095002000L);
+
+        byte[] encodedMSO = new MobileSecurityObjectGenerator("SHA-256",
+                "org.iso.18013.5.1.mDL", deviceKeyFromVector)
+                .addDigestIDs("org.iso.18013.5.1", generateISODigest())
+                .addDigestIDs("org.iso.18013.5.1.US", generateISOUSDigest())
+                .setValidityInfo(signedTimestamp, validFromTimestamp, validUntilTimestamp, null)
+                .generate();
+
+        MobileSecurityObjectParser.MobileSecurityObject mso = new MobileSecurityObjectParser()
+                .setMobileSecurityObject(encodedMSO).parse();
+
+        Assert.assertEquals("1.0", mso.getVersion());
+        Assert.assertEquals("SHA-256", mso.getDigestAlgorithm());
+        Assert.assertEquals("org.iso.18013.5.1.mDL", mso.getDocType());
+
+        Assert.assertEquals(Set.of("org.iso.18013.5.1", "org.iso.18013.5.1.US"),
+                mso.getValueDigestNamespaces());
+        Assert.assertNull(mso.getDigestIDs("abc"));
+        checkISODigest(mso.getDigestIDs("org.iso.18013.5.1"));
+        checkISOUSDigest(mso.getDigestIDs("org.iso.18013.5.1.US"));
+
+        Assert.assertEquals(deviceKeyFromVector, mso.getDeviceKey());
+        Assert.assertNull(mso.getDeviceKeyAuthorizedNameSpaces());
+        Assert.assertNull(mso.getDeviceKeyAuthorizedDataElements());
+        Assert.assertNull(mso.getDeviceKeyInfo());
+
+        Assert.assertEquals(signedTimestamp, mso.getSigned());
+        Assert.assertEquals(validFromTimestamp, mso.getValidFrom());
+        Assert.assertEquals(validUntilTimestamp, mso.getValidUntil());
+        Assert.assertNull(mso.getExpectedUpdate());
+    }
+
+    @Test
+    @SmallTest
+    public void testFullMSO() {
```

Comment:
Maybe we could make this a take a `String digestAlgorithm`, remove the `@Test` and then have three tests `testFullMSO_Sha256`, `testFullMSO_Sha384`, `testFullMSO_Sha512` that calls this? That way we would test all the code paths for all three digests...